src/nrtk_explorer/app/ui/layout.py

```diff
@@ -0,0 +1,185 @@
+from pathlib import Path
+from trame.ui.quasar import QLayout
+from trame.widgets import quasar
+from trame.widgets import html
+from nrtk_explorer.app import ui
+
+
+def toolbar(reload=None):
+    with quasar.QHeader():
+        with quasar.QToolbar(classes="shadow-4"):
+            quasar.QToolbarTitle("NRTK_EXPLORER")
+            if reload:
+                quasar.QBtn(
+                    "Reload",
+                    click=(reload,),
+                    flat=True,
+                )
+
+
+def parse_dataset_dirs(datasets):
+    return [{"label": Path(ds).name, "value": ds} for ds in datasets]
+
+
+def parameters(dataset_paths=[], embeddings_app=None, filtering_app=None, transforms_app=None):
+    with html.Div(classes="q-pa-md q-gutter-md"):
+        (
+            dataset_title_slot,
+            dataset_content_slot,
+            _,
+        ) = ui.card("collapse_dataset")
+
+        with dataset_title_slot:
+            html.Span("Dataset Selection", classes="text-h6")
+
+        with dataset_content_slot:
+            quasar.QSelect(
+                label="Dataset",
+                v_model=("current_dataset",),
+                options=(parse_dataset_dirs(dataset_paths),),
+                filled=True,
+                emit_value=True,
+                map_options=True,
+                dense=True,
+            )
+            quasar.QSlider(
+                v_model=("num_images", 15),
+                min=(0,),
+                max=("num_images_max", 25),
+                disable=("num_images_disabled", True),
+                step=(1,),
+            )
+            html.P(
+                "{{num_images}}/{{num_images_max}} images",
+                classes="text-caption text-center",
+            )
+
+            quasar.QToggle(
+                v_model=("random_sampling", False),
+                dense=False,
+                label="Random selection",
+            )
+
+        (
+            embeddings_title_slot,
+            embeddings_content_slot,
+            embeddings_actions_slot,
+        ) = ui.card("collapse_embeddings")
```

Comment:
Would it make sense to remain this snippet in core.py in the same way that we do a few lines down for `embedding_app.settings_widget(), filtering_app.... transform_app`. It seems harder to follow that `current_dataset`, `num_image_max` ... are `observed/managed` at core.py but used here.

In any case I do not feel too strong about this so I am fine  if you see it fit, ill approve this PR :)

Reply:
I'm not keen on the build_layout externalized but as a first step, I'm fine with it. Ideally, the external stuff should just be classes (not functions) that capture a reusable UI structure. Then in the various app-ui(s), we keep our method and just instantiate those "bigger" ui elements.

Reply:
I see, fair point.